includes/WcfceMailProcessor.inc

```diff
@@ -486,25 +493,37 @@ class WcfceMailProcessor {
    *
    * @param string $email
    *   Email address to search by.
+   * @param int $primary
+   *   (Optional) Search by primary email address or not, possible values:
+   *   1 - search by primary email only,
+   *   0 - search by email other than primary,
+   *   -1 - search by any email.
    *
    * @return int
    *   Civicrm contact id, or 0 if not found.
    *
    * @throws \CiviCRM_API3_Exception
    *   An exception generated by civicrm_api3().
    */
-  private function getContactIdByEmailFromDb($email) {
+  private function getContactIdByEmailFromDb($email, $primary = -1) {
```

Comment:
I think we should make this primary parameter to pass in strings rather than integers, will be more readable if we have the following options
- primary
- non-primary
- all

Then create functions named after what we intend to do that uses this function behind the scenes.